src/Sylius/Component/Core/Promotion/Action/UnitFixedDiscountPromotionActionCommand.php

```diff
@@ -51,12 +63,21 @@ public function execute(PromotionSubjectInterface $subject, array $configuration
             return false;
         }
 
-        $filteredItems = $this->priceRangeFilter->filter(
-            $subject->getItems()->toArray(),
-            array_merge(['channel' => $subject->getChannel()], $configuration[$channelCode]),
-        );
-        $filteredItems = $this->taxonFilter->filter($filteredItems, $configuration[$channelCode]);
-        $filteredItems = $this->productFilter->filter($filteredItems, $configuration[$channelCode]);
+        if (null === $this->compositeFilter) {
+            $filteredItems = $this->priceRangeFilter->filter(
+                $subject->getItems()->toArray(),
+                array_merge(['channel' => $subject->getChannel()], $configuration[$channelCode]),
+            );
+            $filteredItems = $this->taxonFilter->filter($filteredItems, $configuration[$channelCode]);
+            $filteredItems = $this->productFilter->filter($filteredItems, $configuration[$channelCode]);
+        }
+
+        if (null !== $this->compositeFilter) {
+            $filteredItems = $this->compositeFilter->filter(
+                $subject->getItems()->toArray(),
+                array_merge(['channel' => $subject->getChannel()], $configuration[$channelCode]),
+            );
+        }
```

Comment:
Would be nice to have this whole block in a private method, or maybe even place it in the `UnitDiscountPromotionActionCommand` and make it protected.

Reply:
for now i will move it to the private method, let's see, what others would say